src/Microsoft.TestPlatform.Common/Filtering/FilterHelpers.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.VisualStudio.TestPlatform.Common.Resources;
+
+    public static class FilterHelpers
+    {
+        private const char EscapePrefix = '%';  
+        private static readonly Dictionary<char, char> LookUpMap;  
+        private static readonly Dictionary<char, char> ReverseLookUpMap;    
+        private static readonly char[] SpecialCharacters;
+
+        static FilterHelpers()
+        {
+            LookUpMap = new Dictionary<char, char>()
+            {
+                { '%', '0'},
+                { '(', '1'},
+                { ')', '2'},
+                { '&', '3'},
+                { '|', '4'},
+                { '=', '5'},
+                { '!', '6'},
+                { '~', '7'},
+            };
+
+            ReverseLookUpMap = new Dictionary<char, char>();
+            foreach (var kvp in LookUpMap)
+            {
+                ReverseLookUpMap[kvp.Value] = kvp.Key;
+            }
+
+            SpecialCharacters = LookUpMap.Keys.ToArray();
+        }
+
+        /// <summary>
+        /// Escapes a set of special characters for filter (%, (, ), &, |, =, !, ~) by replacing them with their escape sequences. 
+        /// </summary>
+        /// <param name="str">The input string that contains the text to convert.</param>
+        /// <returns>A string of characters with special characters converted to their escaped form.</returns>
+        public static string Escape(string str)
```

Comment:
I am guessing. we are just going to replicate this code on the IDE side. If yes, it is problematic, maintenance issues and what not.
Also, we would need to document this for other consumers of TP to use this escaping logic

Reply:
I made the helper class public, so it can be used by other components. But I'm not sure if this is the right place for public helpers.